diff -Nur a/relay/dhcrelay.c dhcp-4.2.4/relay/dhcrelay.c
--- a/relay/dhcrelay.c	2015-08-21 13:20:26.452646293 -0700
+++ dhcp-4.2.4/relay/dhcrelay.c	2015-08-21 13:21:04.212647244 -0700
@@ -319,7 +319,7 @@
 				agent_relay_mode = discard;
 			} else
 				usage();
-		} else if (!strcmp (argv [i], "-l")) {
+		} else if (!strcmp (argv [i], "-l") && local_family == AF_INET) {
 			add_agent_options = 1;
 			add_rfc3527_suboption = 1;
 			if (++i == argc)
